File: regions/business/taskmanagement/client/react/project-master/src/lib/api-client.ts
// system-client の createApiClient を使用して BFF の CSRF 契約に準拠する
// X-XSRF-TOKEN (Cookie) ではなく X-CSRF-Token (/auth/session JSON) を使用する
import { createApiClient, setCsrfToken } from '@k1s0/system-client';

// BFF /auth/session から取得した CSRF トークンを保持する
// setCsrfToken() で外部から注入する（AuthProvider が /auth/session 後に呼び出す）
export { setCsrfToken };

// BFF API クライアント: CSRF トークン管理と 401 リダイレクトを自動処理する
export const apiClient = createApiClient({
  baseURL: '/bff/api/v1',
  onUnauthorized: () => {
    window.location.href = '/login';
  },
});

File: regions/service/activity/client/flutter/activity/lib/utils/api_client.dart
// system_client の ApiClient を使用して BFF の CSRF 契約に準拠する
// 旧 DioClient は x-csrf-token レスポンスヘッダーを読んでいたが、
// BFF は /auth/session の JSON ボディで csrf_token を返すため誤りだった
// 本ファイルは後方互換のために ApiClient.create へ委譲する
import 'package:dio/dio.dart';
import 'package:system_client/system_client.dart';

/// ApiClient のファクトリメソッドラッパー
/// system_client の ApiClient.create() に委譲し、CSRF 契約を正しく実装する
class ActivityApiClient {
  /// 指定されたベースURLでDioインスタンスを生成する
  /// system_client の ApiClient.create() に委譲する
  static Dio create({required String baseUrl, CsrfTokenProvider? csrfTokenProvider}) {
    return ApiClient.create(
      baseUrl: baseUrl,
      csrfTokenProvider: csrfTokenProvider,
    );
  }
}
